convert2rhel/checks.py

```diff
@@ -318,3 +341,24 @@ def get_unsupported_kmods(host_kmods, rhel_supported_kmods):
     Ignore certain kmods mentioned in the system configs. These kernel modules moved to kernel core, meaning that the
     functionality is retained and we would be incorrectly saying that the modules are not supported in RHEL."""
     return host_kmods - rhel_supported_kmods - set(system_info.kmods_to_ignore)
+
+
+def check_rhel_compatible_kernel_is_used():
+    """Check if the kernel packages are RHEL compatible."""
+    kernel_release, _ = run_subprocess("uname -r")
+    kernel_release = kernel_release.strip()
+    if not RHEL_COMPATIBLE_KERNEL_RELEASE_RE.match(kernel_release):
```

Comment:
this is very wierd check to be honest. it's actually not checking whether kernel is comparible or not, it's checking just particular used (NE)VRA format. Installing kernel of different version on the machine it's easy to pass the test. I even found some kernels on centos that do not match the regexp (realtime kernel e.g.) and are RHEL compatible - but the most possibly it's not supported for the conversion. The *minimal* change you should do is
- re-wording
- extend the condition to check the kernel version is the expected one for the RHEL X system

it's just suggestion guys.  I would even rather replace RHEL_COMPATIBLE_KERNEL_RELEASE_RE in your case completely to be honest, as this seems to me totaly wrong idea.

Reply:
Path to RT repo for centos 7.9 for some examples: http://mirror.centos.org/centos/7.9.2009/rt/x86_64/Packages/

Reply:
I agree this is quite a simplistic approach but I know of no easy way how to ascertain the kernel compatibility.
I can imagine we could run repoquery over the RHEL repos to see if the booted kernel is available there. But that would be limiting too much.
When you say you'd replace the RHEL_COMPATIBLE_KERNEL_RELEASE_RE, what would you replace it with?

The realtime use case is something I was not aware of. But convert2rhel does not consider this use case in other ways as well (standard kernel is being installed). Making sure the rt use is covered would be a new feature to work on.

Reply:
@pirat89, is this wording more fitting: https://github.com/oamg/convert2rhel/pull/250#discussion_r622442288 ?

Reply:
From my experience, I would do something like this (pseudocode):
```suggestion
    # { el_major_version : kernel_version }
    EXPECTED_KERNEL_VERSIONS = {
        "6": "2.6.32"
        "7" : "3.10.0"
        "8": "4.18.0"
    }
    version, release = run_subprocess("uname -r")[0].strip().split("-")
    if EXPECTED_KERNEL_VERSIONS[sys_major_version] != version or any([i in release for i in ["rt", "uek"]])
```

Reply:
ideally I would create is_realtime_kernel, is_unbreakable_kernel functions and provide better msg based on which of those it is. but that's just a nitpick.

Reply:
With the `EXPECTED_KERNEL_VERSIONS`, would we have to update this every time there is a release?

Reply:
As I understood - not. For the given major version of the CentOS or Oracle Linux, there is not changed first three numbers of the kernel version.
/cc @pirat89 @bocekm

Reply:
In this I trust @pirat89 who says that there's no kernel version change within one RHEL major version.